.github/workflows/rust.yml

```diff
@@ -60,14 +60,27 @@ jobs:
   test:
     needs: diff
     if: needs.diff.outputs.isRust == 'true'
-    runs-on: [self-hosted, self-hosted-ubuntu]
     timeout-minutes: 30
+    runs-on: ${{ matrix.os }}
     strategy:
+      matrix:
+        os:
+          - [self-hosted, self-hosted-ubuntu]
+          - [self-hosted, windows]
       fail-fast: false
     steps:
+      # Set the path for windows builds: the runner needs to be able to find
+      # llvm binaries as well as the "bash" binary included with msys64.
+      - name: Adjust path for windows builds
+        if: "startsWith(runner.os, 'windows')"
+        run: |
+          echo "C:\Program Files\llvm\bin;C:\msys64\usr\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```

Comment:
I can actually add these to the system path for the build user, which is probably better.

Reply:
Yeah anything we can do on the machine itself to make the jobs as identical as possible would be awesome.

Reply:
Added to system path and removed